lib/source_opam_file.ml

```diff
@@ -0,0 +1,208 @@
+open Import
```

Comment:
`Source_opam_file` is the name for opam files in the repositories sources, as opposed to a file that we would generate like the lockfile.

I think this could be used to factor out more code used to interpret them, not just the extra solver configuration bits, hence the name. I'm happy to change to a better name if it conveys the same idea or to make it a configuration specific module if you prefer but I quite liked having such a module.